keps/sig-api-machinery/1965-kube-apiserver-identity/README.md

```diff
@@ -117,12 +143,30 @@ flag.
 
 ### Test Plan
 
-  - integration test for creating the Namespace and the Lease on kube-apiserver
-    startup
-  - integration test for not creating the StorageVersions after creating the
-    Lease
-  - integration test for garbage collecting a Lease that isn't refreshed
-  - integration test for not garbage collecting a Lease that is refreshed
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `staging/src/k8s.io/apiserver/pkg/endpoints`
+
+##### Integration tests
+
+[apiserver_identity_test.go](https://github.com/kubernetes/kubernetes/blob/24238425492227fdbb55c687fd4e94c8b58c1ee3/test/integration/controlplane/apiserver_identity_test.go)
+- integration test for creating the Namespace and the Lease on kube-apiserver startup
+- integration test for not creating the StorageVersions after creating the Lease
+- integration test for garbage collecting a Lease that isn't refreshed
+- integration test for not garbage collecting a Lease that is refreshed
+
+##### e2e tests
+
+Proposed e2e tests:
+- an e2e test that validates the existence of the Lease objects per kube-apiserver
+- an e2e test that restarts a kube-apiserver and validates that a new Lease is created
```

Comment:
not sure how a standard e2e test could do this.

Reply:
> not sure how a standard e2e test could do this.

Oh, possibly an upgrade test could since at the end of the test you could compare against the starting state.

Reply:
Thinking of re-using something similar to [RestartControllerManager](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/kubesystem/kubesystem.go#L30), but for kube-apiserver

Reply:
I'm willing to do either or both.